drivers/usb/device/usb_dc_stm32.c

```diff
@@ -36,6 +37,23 @@ LOG_MODULE_REGISTER(usb_dc_stm32);
 #error "Only one interface should be enabled at a time, OTG FS or OTG HS"
 #endif
 
+/*
+ * Some STM32U5xx parts support a USB HS PHY which is clocked by the HSE clock or PLL1_P
+ * clock directly.  This requires specific configuration.
+ */
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_otghs) && defined(CONFIG_SOC_SERIES_STM32U5X)
+
+#define IS_EQ_PROP(inst, prop, compare_value) IS_EQ(DT_PROP(inst, prop), compare_value)
+#if DT_NODE_HAS_PROP(DT_PHANDLE_BY_IDX(DT_NODELABEL(usbotg_hs), phys, 0), clock_cfg)
+#define OTG_HS_PHY_REFERENCE_CLOCK                                                                 \
+	_CONCAT(SYSCFG_OTG_HS_PHY_CLK_SELECT_,                                                     \
+		DT_PROP(DT_PHANDLE(DT_NODELABEL(usbotg_hs), phys), clock_cfg))
+#else
+#define OTG_HS_PHY_REFERENCE_CLOCK SYSCFG_OTG_HS_PHY_CLK_SELECT_1
+#endif
```

Comment:
There are various ways to provide a default value:
- Pre processor Macros : `COND_CODE_1(DT_NODE_HAS_PROP`: See examples https://github.com/search?q=repo%3Azephyrproject-rtos%2Fzephyr%20COND_CODE_1(DT_NODE_HAS_PROP&type=code)
- [DT Bindings default](https://docs.zephyrproject.org/latest/build/dts/bindings-syntax.html#default), which  I guess would be appropriate and the most elegant in this case.

Reply:
Indeed, this is even better as this depends on each board implementation.